### PR TITLE
feat: shift + arrow up/down to select multiple values

### PR DESCRIPTION
**Pull request by LiKang6688**

### CODE REVIEW

**Comment by haxxmaxx** [comment]
I am doing some refactoring for the selection-utils for performance so this will have to wait a bit

File: src/table/components/TableBodyWrapper.jsx [hunk 1]
```diff
@@ -98,6 +98,8 @@ function TableBodyWrapper({
                       selectionState,
                       cell,
                       selectionDispatch,
+                      tableRows: rows,
```
Comment (haxxmaxx) [inline src/table/components/TableBodyWrapper.jsx:101]: I think this is not a good idea. You are basically putting all the table data on every cell. I see you need qElemNumber for the cell above and bellow. If you have to pass multiple cells, I would prefer only sending those two cells extra cells that are needed

Comment (LiKang6688) [reply src/table/components/TableBodyWrapper.jsx:101]: I agree that putting all the table data on every cell is not good. It is a bit overkill and would be good to pass one-column data.

But putting all the table data on every cell does not impact performance. It is a copy by reference.

And getting two cells extra cells for every cell impacts performance. As it needs to do operations on the array.

File: src/table/utils/selections-utils.js [hunk 2]
```diff
@@ -95,7 +104,17 @@ export function selectCell({ selectionState, cell, selectionDispatch, evt, annou
     return;
   }
 
-  selectedRows = getSelectedRows({ selectedRows, qElemNumber, rowIdx, evt });
+  if (isSelectMultiCells) {
```
Comment (haxxmaxx) [inline src/table/utils/selections-utils.js:107]: should be able to check what keys are pressed on `evt` and get rid of this argument

File: src/table/components/TableBodyWrapper.jsx (hunk 1 above)
Comment (haxxmaxx) [reply src/table/components/TableBodyWrapper.jsx:101]: what about putting the previous and next cell `qElemNumber` on the cell in `handleData`? That way you don't have to add any arguments at all.

**Comment by LiKang6688** [comment]
a new one gonna come.

**Pull request closed without merging**